libobs/obs.c

```diff
@@ -2019,6 +2019,26 @@ obs_source_t *obs_get_source_by_name(const char *name)
 				   obs_source_addref_safe_);
 }
 
+obs_source_t *obs_get_source_by_uuid(const char *uuid)
+{
+	struct obs_source **first = &obs->data.first_source;
+	struct obs_source *source;
+
+	pthread_mutex_lock(&obs->data.sources_mutex);
+
+	source = *first;
+	while (source) {
+		if (strcmp(source->context.uuid, uuid) == 0) {
+			source = obs_source_addref_safe_(source);
+			break;
+		}
+		source = (void *)source->context.next;
```

Comment:
Shouldn't this be cast to `obs_source_t *` or `struct obs_source *`? It's inconsistent across libobs, but `obs_source_t *` seems to be more common.

Reply:
Yeah it is inconsistent, this is essentially the same as `obs_get_transition_by_name` immediately below it, which uses `void *`, but I can change it to be clearer.

Reply:
Updated to use `struct obs_source *` now. Seems we generally don't use the typedef'd versions here, but as you said, it's not consistent.